core/blockchain/impl/block_tree_impl.cpp

```diff
@@ -195,8 +201,11 @@ namespace kagome::blockchain {
     }
 
     // Prepare and create block tree basing last finalized block
-    auto tree = std::make_shared<TreeNode>(
-        last_finalized_block_info.hash, last_finalized_block_info.number, true);
+    auto tree = std::make_shared<TreeNode>(last_finalized_block_info.hash,
+                                           last_finalized_block_info.number,
+                                           nullptr,
```

Comment:
Nullptr in args looks stinky. How about to use optional?

Reply:
No, it was that way before (there was no argument, so field was not initialized)

Reply:
Yes, I agree. It is technical debt and we can't resolve its urgently, but we can keep it down.